frb_example/benches/rust/src/executor.rs

```diff
@@ -0,0 +1,237 @@
+        maybe_subscribe(json);
+        Self {
+            inner: ThreadPoolExecutor::new(error_handler),
+            json,
+        }
+    }
+}
+
+impl Executor for BenchExecutor {
+    /// wrap [`bench_around`] over async FFI calls
+    fn execute<TaskFn, TaskRet>(&self, wrap_info: WrapInfo, task: TaskFn)
+    where
+        TaskFn: FnOnce(TaskCallback) -> anyhow::Result<TaskRet> + Send + UnwindSafe + 'static,
+        TaskRet: IntoDart,
+    {
+        let debug_name_string = wrap_info.debug_name.to_string();
+        let json = self.json;
+        self.inner.execute(wrap_info, move |task_callback| {
+            Self::bench_around(&debug_name_string, json, move || task(task_callback))
+        })
+    }
+
+    /// wrap [`bench_around`] over sync FFI calls
+    fn execute_sync<SyncTaskFn, TaskRet>(
+        &self,
+        wrap_info: WrapInfo,
+        sync_task: SyncTaskFn,
+    ) -> anyhow::Result<flutter_rust_bridge::SyncReturn<TaskRet>>
+    where
+        flutter_rust_bridge::support::WireSyncReturnData: From<TaskRet>,
+        SyncTaskFn: FnOnce() -> anyhow::Result<flutter_rust_bridge::SyncReturn<TaskRet>>
+            + std::panic::UnwindSafe,
+    {
+        let debug_name_string = wrap_info.debug_name.to_string();
+        self.inner.execute_sync(wrap_info, move || {
+            Self::bench_around(&debug_name_string, self.json, sync_task)
+        })
+    }
+}
+
+/// executor (on native platforms)
+///
+/// use [tracing](https://docs.rs/crate/tracing/0.1.36) for stdout
+/// or record metrics as json
+#[cfg(not(target_family = "wasm"))]
+impl BenchExecutor {
+    /// benchmark every bridge-wired function call (on native platforms)
+    ///
+    /// - smallest time unit in rust is _nanoseconds_
+    ///   > note that at this point we're out of dart vm, whose smallest time unit is _microseconds_
+    /// - underlying implementation relies on standard [Instant](https://doc.rust-lang.org/std/time/struct.Instant.html)
+    fn bench_around<F, R>(bench_name: &str, json: bool, f: F) -> R
+    where
+        F: FnOnce() -> R,
+    {
+        if !json {
+            use tracing::{span, trace, Level};
+            span!(Level::TRACE, "frb-executor");
+            trace!("(Rust) execute [{}] start", bench_name);
+        }
+        let start = std::time::Instant::now();
+        let ret = f();
+        let elapsed = start.elapsed().as_nanos();
+        if !json {
+            use tracing::{span, trace, Level};
+            span!(Level::TRACE, "frb-executor");
+            trace!(
+                "(Rust) execute [{}] end delta_time={}{}",
+                bench_name,
+                elapsed,
+                Unit::Nanoseconds.acronym()
+            );
+        } else {
+            record(bench_name, elapsed as u64, Unit::Nanoseconds);
+        }
+        ret
+    }
+}
+
+/// executor (on web platform)
+///
+/// use [console](web_sys::console) a.k.a [console](https://developer.mozilla.org/en-US/docs/Web/API/console) for stdout
```

Comment:
@Desdaemon I remembered, when implementing web support, you did something similar to console.log. What should we use to log the messages - console::log or others? B/c I hope the codebase is consistent and do not have two logging mechanism

Reply:
I met a couple of issues on this, hence why I landed on web_sys console.
It's either piping logs to dart with a streamsink, and that's cumbersome, either nothing will appear since rust logs are hidden away in wasm.

Reply:
I also looked into [instant](https://crates.io/crates/instant) as I stated in the docs (see `troubleshooting` section at the bottom in the book) but IIRC then there's something like you depend on `stdweb` which doesn't play well with `wasm-pack` and the `no-modules` arg that is used in FRB.

Reply:
Yes I think that is reasonable. Just to confirm whether we should, e.g. change code in other parts of frb to be consistent